### PR TITLE
[ATen] Add CPU fp16 support for nll_loss and cross_entropy_loss

### PR DESCRIPTION
**Pull request by etaf**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #123256

Add CPU FP16 support for nll_loss and cross_entropy_loss.
Resolve issue #123328.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Review by jgong5 — APPROVED** [review]
Do we have UT?

**Comment by etaf** [comment]
> Do we have UT?

I'v tried, but not find any existing test case for `nll_loss_out_frame` in Pytorch.

**Review by EikanWang — APPROVED** [review]
Could you help add UT to `test_ops.py`? So it could help XPU as well.

**Comment by etaf** [comment]
> Do we have UT?

UT added.

**Review by EikanWang — APPROVED** [review]
LGTM.

**Comment by EikanWang** [comment]
@etaf , please fix the failures in terms of the sample input.

**Review by jgong5 — CHANGES_REQUESTED** [review]
UT failing...

**Review by EikanWang — APPROVED** [review]

**Comment by EikanWang** [comment]
@jgong5 , may I know if you have any additional comments?

**Review by jgong5 — APPROVED** [review]

**Comment by etaf** [comment]
@mruberry @kulinseth @BowenBao @abock @thiagocrepaldi @wschin 
Hello, can any one help to have a review?

**Review by malfet — APPROVED** [review]

File: test/test_mps.py [hunk 1]
```diff
@@ -987,7 +989,10 @@ def mps_ops_modifier(ops):
         'nn.functional.avg_pool2d': [torch.float16],
         # input types 'tensor<f32>' and 'tensor<1xf16>' are not broadcast compatible
         # Refer to the issue please: https://github.com/pytorch/pytorch/issues/124252
-        'nn.functional.binary_cross_entropy': [torch.float16]
+        'nn.functional.binary_cross_entropy': [torch.float16],
+
+        'nn.functional.nll_loss': [torch.float16],
```
Comment (malfet) [inline test/test_mps.py:994]: @etaf in the future, please add a comment why are you adding a function to a skip-list (and create an issue)
I though the reason for the skip were that it crashed the runtime, but looks like it just it needed a precision tolerance tweak

Comment (etaf) [reply test/test_mps.py:994]: @malfet thanks for your reminder,  I'll take care of it next time。

**Pull request closed without merging**